### PR TITLE
do not run codecov upload step on dependabot PR

### PR DESCRIPTION
**Pull request by NickLucche**
Addressing https://issues.redhat.com/browse/RHOAIENG-12847. Dependabot PRs can't auto-merge because codecov update will fail, as PRs from forks are not allowed to access secrets (to avoid exposure).

## Description

Have a separate PR workflow that checks whether it's coming from dependabot. Should be safe from secrete leaking as other steps (the ones shared with all contributors) do NOT have the codecov secret injected.

### CODE REVIEW

**Comment by dtrifiro** [comment]
Thinking about it, it probably makes more sense to just skip the code coverage upload  step for dependabot PRs, since they won't involve code changes.

**Comment by NickLucche** [comment]
In light of https://docs.codecov.com/docs/codecov-tokens#uploading-without-a-token, I think we can keep it simple as you suggested

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -75,6 +75,7 @@ jobs:
 
       - name: Upload coverage report
         uses: codecov/codecov-action@v4
+        if: github.actor != 'dependabot[bot]'
```
Comment (dtrifiro) [inline .github/workflows/tests.yaml:78]: :100:

**Comment by dtrifiro** [comment]
Looks great, can you update the PR title aswell? Thanks!

**Review by dtrifiro — APPROVED** [review]

**Pull request merged**